app/src/main/kotlin/com/mapzen/erasermap/presenter/MainPresenterImpl.kt

```diff
@@ -166,4 +168,20 @@ public class MainPresenterImpl() : MainPresenter {
         mainViewController?.setMapTilt(MainPresenter.ROUTING_TILT)
         mainViewController?.setMapRotation(Math.toRadians(instruction.bearing.toDouble()).toFloat())
     }
+
+    override fun onPause(mapzenLocation: MapzenLocation?) {
+        if (isRouting() || isRoutingDirectionList()) {
+            return
+        }
+
+        mapzenLocation?.disconnect()
```

Comment:
if not routing AND not routing direction list would do the same thing in 3 lines